documentation/modules/exploit/multi/http/apache_solr_backup_restore.md

```diff
@@ -0,0 +1,279 @@
+## Vulnerable Application
+
+Apache Solr from 6.0.0 through 8.11.2, from 9.0.0 before 9.4.1 is affected by an Unrestricted Upload of File
+with Dangerous Type vulnerability which can result in remote code execution in the context of the user running
+Apache Solr. When Apache Solr creates a Collection, it will use a specific directory as the classpath and load
+some classes from it. The backup function of the Collection can export malicious class files uploaded by
+attackers to the directory, allowing Solr to load custom classes and create arbitrary Java code. Execution
+can further bypass the Java sandbox configured by Solr, ultimately causing arbitrary command execution.
+
+### Setup
+
+Install a vulnerable instance of Apache Solr with the following docker-compose file. The instance must be running in 
+"Cloud mode" in order to be vulnerable which is why the `-c` argument is included in the `solr start` command.
+
+#### Apache Solr 9.0.0 (no Authentication)
+```yml
+version: '3'
+
+services:
+  solr:
+    image: solr:9.0.0
+    ports:
+      - "8983:8983"
+      - "5005:5005"
+    command: sh -c "solr start -c -a '-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005' && tail -f /dev/null"
```

Comment:
After running `docker-compose` I'm hitting `ms-1    | [-] Failed to connect to the Postgres data service: could not translate host name "db" to address: Name does not resolve` - Any idea what's going on there?

Reply:
Hey @zgoldman-r7, not exactly sure what's going on there. Just to confirm, you're running `docker-compose up` in the same directory as the `docker-compose.yml`? 

It's interesting the error pertains to Postgres. Maybe the issue has something to do with your environment, docker/ docker-compose version or possibly existing configurations/ containers. I just retested this container on a fresh Ubuntu 22.04 and it seemed to work without any issues:

```
msfuser@msfuser-virtual-machine:~$ uname -a
Linux msfuser-virtual-machine 6.5.0-28-generic #29~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Apr  4 14:39:20 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
msfuser@msfuser-virtual-machine:~$ docker-compose -version
docker-compose version 1.29.2, build unknown
msfuser@msfuser-virtual-machine:~$ docker -v
Docker version 24.0.5, build 24.0.5-0ubuntu1~22.04.1
msfuser@msfuser-virtual-machine:~$
msfuser@msfuser-virtual-machine:~$ cat docker-compose.yml
version: '3'

services:
  solr:
    image: solr:9.0.0
    ports:
      - "8983:8983"
      - "5005:5005"
    command: sh -c "solr start -c -a '-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005' && tail -f /dev/null"
msfuser@msfuser-virtual-machine:~$ sudo docker-compose up
[sudo] password for msfuser:
Creating network "msfuser_default" with the default driver
Pulling solr (solr:9.0.0)...
9.0.0: Pulling from library/solr
3c67549075b6: Pull complete
b3cda026f111: Pull complete
d0316070ed7b: Pull complete
5f6a8ff6cd7e: Pull complete
fb9ffccab5f5: Pull complete
6fc2e145c3fe: Pull complete
f345603f3fa0: Pull complete
9f10439f23a4: Pull complete
9c3461a5022e: Pull complete
a14f01cd9e96: Pull complete
4f4fb700ef54: Pull complete
Digest: sha256:5103aea19a40a3462f612cd816d8ba6d0f58ff3ff5bf3930298393068797e83d
Status: Downloaded newer image for solr:9.0.0
Creating msfuser_solr_1 ... done
Attaching to msfuser_solr_1
solr_1  | Warning: Available entropy is low. As a result, use of the UUIDField, SSL, or any other features that require
solr_1  | RNG might not work properly. To check for the amount of available entropy, use 'cat /proc/sys/kernel/random/entropy_avail'.
solr_1  |
solr_1  | Waiting up to 180 seconds to see Solr running on port 8983 [-]
solr_1  | Started Solr server on port 8983 (pid=59). Happy searching!
solr_1  |
```

It's also working on my MacOS host with the following docker versions:
```
➜  apache_solr docker-compose -v
Docker Compose version v2.24.6-desktop.1
➜  apache_solr docker -v
Docker version 25.0.3, build 4debf41
➜  apache_solr
➜  apache_solr docker-compose up
[+] Running 1/0
 ✔ Container apache_solr-solr-1  Created                                                                                                                                                                                               0.0s
Attaching to solr-1
solr-1  | Warning: Available entropy is low. As a result, use of the UUIDField, SSL, or any other features that require
solr-1  | RNG might not work properly. To check for the amount of available entropy, use 'cat /proc/sys/kernel/random/entropy_avail'.
solr-1  |
solr-1  | Waiting up to 180 seconds to see Solr running on port 8983 [\]
solr-1  | Started Solr server on port 8983 (pid=61). Happy searching!
solr-1  |
```

Reply:
Thanks for taking a look at the PR! Let me know if testing on Ubuntu works or if you have any other questions :)